Support specifying scales in preprocessing div2k dataset

wileechou opened the pull request:
### Motivation
We want build a data set like DIV2KRK which contain low-quality(LQ) images, ground-truth(GT) images and blur kernels and use it for training models like Deep Alternating Network(DAN), which means we also need to crop them into sub images. This PR create a scripts to generate this data set.

### Modification
We create a new file` preprocess_div2k_dataset_bsr.py`, a copy of `preprocess_div2k_dataset.py`, and then add a function to generate blur kernel according to configuration, and we no longer use DIV2K LR images to crop into sub images but apply pipeline used in DAN's paper: **GT-->blur-->down-sampling-->LQ**, and we also stored corresponding blur kernel in `.mat` format using same base name.

ckkelvinchan commented:
Hello, thank you for your contribution. May I ask why do you pre-generate the LR images instead of generating the LR during training? I think the degradations could be more diverse if we apply the degradations during training.

wileechou commented:
> Hello, thank you for your contribution. May I ask why do you pre-generate the LR images instead of generating the LR during training? I think the degradations could be more diverse if we apply the degradations during training.

Yes, you're right.  Maybe I think pre-generating low-quality images is  convenient and intuitive,  and if we generate lq images during training, we will no longer need LR sub images generated by `preprocess_div2k_dataset.py`. 

Thanks for your advice! I have a new idea. We add a new argument in `preprocess_div2k_dataset.py`  to control whether generating LR sub images or not as more and more researches are performing more complex modeling of the degradation process, not just down-sampling.  And we can degrade images during training.

wileechou commented:
I have modified my pull request by

1. Removing `preprocess_div2k_dataset_bsr.py`
2. Adding a new argument `--custom-degradation` into 'preprocess_div2k_dataset.py' to control whether crop bicubic LR images
3. Adding some illustrations into `README.md` and `README_zh-CN.md`

ckkelvinchan commented:
> I have modified my pull request by
> 
> 1. Removing `preprocess_div2k_dataset_bsr.py`
> 2. Adding a new argument `--custom-degradation` into `preprocess_div2k_dataset.py` to control whether crop bicubic LR images
> 3. Adding some illustrations into `README.md` and `README_zh-CN.md`

I think you can simply use an argument `scale` to specify whether you want to process the downsampled images. 

If `scale = []`, then only the HR images will be processed.

wileechou commented:
> > I have modified my pull request by
> > 
> > 1. Removing `preprocess_div2k_dataset_bsr.py`
> > 2. Adding a new argument `--custom-degradation` into `preprocess_div2k_dataset.py` to control whether crop bicubic LR images
> > 3. Adding some illustrations into `README.md` and `README_zh-CN.md`
> 
> I think you can simply use an argument `scale` to specify whether you want to process the downsampled images.
> 
> If `scale = []`, then only the HR images will be processed.

Thanks for your advice. I have updated it.

ckkelvinchan commented on tools/data/super-resolution/div2k/preprocess_div2k_dataset.py:
```diff
@@ -22,6 +22,7 @@ def main_extract_subimages(args):
             A higher value means a smaller size and longer compression time.
             Use 0 for faster CPU decompression. Default: 3, same in cv2.
 
+        scale (List[int]): scale factor
```
scale (list[int]): The downsampling factors corresponding to the LR folders you want to process. Default: [2, 3, 4].

ckkelvinchan commented on tools/data/super-resolution/div2k/preprocess_div2k_dataset.py:
```diff
@@ -31,7 +32,7 @@ def main_extract_subimages(args):
     Usage:
         For each folder, run this script.
-        Typically, there are four folders to be processed for DIV2K dataset.
+        Typically, there are four folders to be processed for DIV2K dataset according to scale factor list ([2,3,4])
```
By default, there are four folders to be processed for DIV2K dataset according to scale factor list ([2,3,4])
  DIV2K_train_HR
  DIV2K_train_LR_bicubic/X2
  DIV2K_train_LR_bicubic/X3
  DIV2K_train_LR_bicubic/X4
After processing, each sub_folder should have the same number of subimages.
You can also process a subset of folders by modifying the argument `scale`. Remember to modify opt configurations according to your settings.

(May need to format it)

OpenMMLab-Coodinator commented:
Hi @wileewang ！First of all, we want to express our gratitude for your significant PR in this project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you are Chinese or have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤
